### ISSUE TITLE
Debian based for rkt and ipam: dhcp

### ISSUE THREAD

**Issue by JulienBalestra** [containernetworking/cni#253]
Hi everyone, 

I plan to use rkt over Debian and I'm face to an issue.
I'd like to use external already existing DHCP service for the rkt containers.
So I follow the documentation and I write the config file 

```
cat << EOF > /etc/rkt/net.d/10-mynet.conf
{
    "name": "mynet",
    "type": "bridge",
    "bridge": "cni0",
    "isGateway": true,
    "ipMasq": true,
    "ipam": {
        "type": "dhcp"
    }
}
EOF
```

I also have the dhcp's cni program running: `./dhcp daemon `
Process running `root      6157     1  0 Jun20 ?        00:00:00 ./dhcp daemon` with the associate socket: 
```
ls -ls  /run/cni/dhcp.sock 
0 srwxr-xr-x 1 root root 0 Jun 20 17:57 /run/cni/dhcp.sock
```

And now basically trying to start the container: 
`rkt  run  --debug --net=mynet --interactive quay.io/julienbalestra/kafka --exec /bin/bash`

This command give me the following output:
```
image: using image from local store for image name coreos.com/rkt/stage1-coreos:1.6.0
image: using image from local store for image name quay.io/julienbalestra/zookeeper
stage0: Preparing stage1
stage0: Writing image manifest
stage0: Loading image sha512-9e10832377eacc6a2017f7f7bdb872c41dbff97d73b6ac20d843ad7d3cd99a25
stage0: Writing image manifest
stage0: Writing pod manifest
stage0: Setting up stage1
stage0: Wrote filesystem to /var/lib/rkt/pods/run/14948ec4-80dd-4175-a980-acddeb92acad
stage0: Pivoting to filesystem /var/lib/rkt/pods/run/14948ec4-80dd-4175-a980-acddeb92acad
stage0: Execing /init
networking: loading networks from /etc/rkt/net.d
networking: loading network mynet with type bridge
stage1: failed to setup network
  ������error adding network "mynet"
    ������error calling DHCP.Allocate: path "/var/lib/rkt/pods/run/14948ec4-80dd-4175-a980-acddeb92acad/netns" doesn't contain "ns/"
```

I try other plugin like ptp and macvlan, they all gave me the same result.
Noticed the other ipam type like host-local works fine.

Going to the code, I found the following clues: 
https://github.com/containernetworking/cni/blob/0136a73037a0bf57c7292448dd03947ca31c3c0f/pkg/ns/ns.go#L110 apparently lead by my kernel version:
```
uname -a
Linux statefull-gi1m3du1bw 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt25-2 (2016-04-08) x86_64 GNU/Linux
```
It's using the PROCFS_MAGIC define who's apparently need `ns/`...

Anyone can give me more pointers like https://github.com/containernetworking/cni/blob/0136a73037a0bf57c7292448dd03947ca31c3c0f/pkg/ns/ns.go#L81 for better understanding of the requirements for my usecase ?

**Comment by JulienBalestra** [comment]
Missing the rkt version: 1.6.0

**Comment by JulienBalestra** [comment]
Fixed by rkt 1.8.0.
